go/porcelain/deploy.go

```diff
@@ -401,8 +404,9 @@ func (n *Netlify) WaitUntilDeployLive(ctx context.Context, d *models.Deploy) (*m
 	return n.waitForState(ctx, d, "ready")
 }
 
-func (n *Netlify) uploadFiles(ctx context.Context, d *models.Deploy, files *deployFiles, observer DeployObserver, t uploadType, timeout time.Duration) error {
+func (n *Netlify) uploadFiles(ctx context.Context, d *models.Deploy, files *deployFiles, observer DeployObserver, t uploadType, timeout time.Duration, skipRetry bool) error {
 	sharedErr := &uploadError{err: nil, mutex: &sync.Mutex{}}
+	permanentErr := &backoff.PermanentError{Err: nil}
```

Comment:
It's a bit unclear for me why we instantiate `permanentErr` here, as could you elaborate?

as far as I know `backoff.PermanentError` is to be used as return from a `backoff.Retry()` block

Reply:
Initially we weren't sure how to use PermanentError, but your other comment pointed us in the right direction I think